synapse/server.py

```diff
@@ -403,7 +404,7 @@ def get_room_shutdown_handler(self) -> RoomShutdownHandler:
         return RoomShutdownHandler(self)
 
     @cache_in_self
-    def get_sendmail(self) -> sendmail:
+    def get_sendmail(self) -> Callable[..., defer.Deferred]:
```

Comment:
I started adding a better type signature here, but `sendmail` is fairly complicated and can take bytes and str as almost every argument.